fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java

```diff
@@ -198,7 +199,8 @@ public LoanTransaction makeRepayment(final LoanTransactionType repaymentTransact
                 defaultLoanLifecycleStateMachine, existingTransactionIds, existingReversedTransactionIds, isRecoveryRepayment,
                 scheduleGeneratorDTO, isHolidayValidationDone);
 
-        saveLoanTransactionWithDataIntegrityViolationChecks(newRepaymentTransaction);
+        newRepaymentTransaction = saveLoanTransactionWithDataIntegrityViolationChecks(newRepaymentTransaction);
+        newRepaymentTransaction.setLoanChargesPaid(loanChargePaidByRepository.findByLoanTransaction(newRepaymentTransaction));
```

Comment:
I dont think we need this. The chargePaidBy associations are already assigned to the loanTransaction

Reply:
you're right. it's removed